fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -150,7 +170,11 @@ public LuceneIndexMaintainer(@Nonnull final IndexMaintainerState state, @Nonnull
                         document.add(new StoredField(PRIMARY_KEY_FIELD_NAME, ref));
                         document.add(new SortedDocValuesField(PRIMARY_KEY_SEARCH_NAME, ref));
                         for (int i = 0; i < fieldNames.size(); i++) {
-                            document.add(new TextField(fieldNames.get(i), entry.getKey().getString(i), Field.Store.NO));
+                            String value = entry.getKey().getString(i);
+                            if (value == null) {
+                                value = "";
+                            }
```

Comment:
Is this better than not adding it to the document? What are the differences in terms of query behavior? Ignoring 3VL, can't I write `NOT x` to exclude documents where `x` is set?

Reply:
thats a good point. I think that would be the preferred behavior yes. I'll write a few tests to check it out.

Reply:
I accommodated for this in the planner. If we change it then we can change it there too.